torch/distributions/multinomial.py

```diff
@@ -0,0 +1,87 @@
+import torch
+from torch.distributions.distribution import Distribution
+from torch.distributions import Categorical
+from numbers import Number
+from torch.distributions import constraints
+from torch.distributions.utils import log_sum_exp
+
+
+class Multinomial(Distribution):
+    r"""
+    Creates a Multinomial distribution parameterized by `probs` and `total_count`.
+
+    If `probs` is 1D with length-`K`, each element is the relative probability
+    of sampling the class at that index.
+
+    If `probs` is 2D, it is treated as a batch of probability vectors.
+
+    Example::
+
+        >>> m = Multinomial(10, torch.Tensor([ 1, 1, 1, 1]))
+        >>> m.sample()  # equal probability of 0, 1, 2, 3
+         2
+         2
+         3
+         3
+        [torch.FloatTensor of size 4]
+
+
+    Args:
+        total_count (Tensor or Variable): number of trials
```

Comment:
Could you add a comment above indicating that
1. `total_count` need not be specified if only `.log_prob()` is called
2. `.sample()` requires a single shared `total_count` for all parameters and samples, but
3. `.log_prob()` allows different `total_count` for each parameter and sample.

One perspective on `torch.distributions` is that it is a handy collection of methods that are packaged as flyweight classes. From that perspective, you've implemented a very handy heterogenous-count `.log_prob()` method, even thought it doesn't quite fit into the flyweight package. It's nice to note that. BTW @martinjankowiak and I are finding this method very handy in implementing LDA where different documents in a batch may have different total word counts.